bat-utils/lib/runtime-wallet.js

```diff
@@ -475,4 +496,118 @@ Wallet.providers.mock = {
 }
 Wallet.providers.mockHttpSignature = Wallet.providers.mock
 
+Wallet.providers.simplex = {
+  purchaseBAT: async function (info, amount, currency, language) {
+    const fiat = 'USD'
+    const now = underscore.now()
+    const params = {
+      currency: 'XBT',
+      partner: 'brave',
+      version: '1'
+    }
+    let expires, quote, rate, result
+
+    if (!this.runtime.config.simplex) return
+
+    if (!this.currency.fiatP(currency)) {
+      rate = underscore.pick(this.currency.rates[currency] || {}, fiat)
+      if (!rate) return
+
+      amount = Math.ceil(this.currency.alt2fiat(currency, amount, fiat, true))
+      currency = fiat
+    }
+
+    quote = info.simplex
+    if ((!quote) || (quote.expires <= now)) {
+      result = await braveHapi.wreck.post(this.runtime.config.simplex.url + '/wallet/merchant/v2/quote', {
+        headers: {
+          accept: 'application/json',
+          authorization: 'ApiKey ' + this.runtime.config.simplex.api_key,
+          'content-type': 'application/json'
+        },
+        payload: JSON.stringify({
+          end_user_id: info.paymentId,
```

Comment:
is this the user wallet's paymentId? do we really need to send that to simplex? can we deterministicly generate an id only used with simplex?

Reply:
we can certainly add another stable UUID. i will do that.